maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java

```diff
@@ -311,13 +311,29 @@ else if ( StringUtils.isNotEmpty( request.getMakeBehavior() ) )
             if ( !resumed )
             {
                 throw new MavenExecutionException( "Could not find project to resume reactor build from: " + selector
-                    + " vs " + projects, request.getPom() );
+                    + " vs " + formatProjects( projects ), request.getPom() );
             }
         }
 
         return result;
     }
 
+    private String formatProjects( List<MavenProject> projects )
+    {
+        StringBuilder projectNames = new StringBuilder();
+        for ( int index = 0, size = projects.size(); index < size; index++ )
+        {
+            MavenProject project = projects.get( index );
```

Comment:
Don't iterate lists with indexes, use foreach or iterators instead.

Reply:
There is additional condition inside loop which appends a colon. This can’t be done without regular for loop as foreach doesn’t give a way to check if fetched element is actually the last one.

Reply:
or use StingUtils.join from Commons-lang3

Reply:
@splatch An iterator gives you that.
@slachiewicz He need custom formatting before the separator.

Reply:
It makes sense to use commons only if they are already part of core dependencies.

Reply:
Great, will update all PRs to rely on it. Thank you very much for suggestion.

Reply:
@slachiewicz @michael-o I checked it and `StringUtils` would allow me to glue collection easily but still I need to custom formatting of elements. Whole idea is to return proper input for `-rf` option. 

For now I switch to iterator without `StringUtils`.